src/core/public/application/types.ts

```diff
@@ -289,12 +330,16 @@ export type AppSearchDeepLink = {
       path: string;
       /** Optional array of links that are 'underneath' this section in the hierarchy */
       searchDeepLinks?: AppSearchDeepLink[];
+      /** Optional meta containing keywords to match with in deep links search */
+      meta?: AppMeta;
     }
   | {
       /** Optional path to access this section. Omit if this part of the hierarchy does not have a page URL. */
       path?: string;
       /** Array links that are 'underneath' this section in this hierarchy. */
       searchDeepLinks: AppSearchDeepLink[];
+      /** Meta containing keywords to match with in deep links search */
+      meta?: AppMeta;
```

Comment:
If no path is provided, then we don't have a 'landing page'/UI to view, so adding meta.keywords here isn't actually needed. However, we need to provide it for the case when we do have a path. Leaving it as optional should be ok.